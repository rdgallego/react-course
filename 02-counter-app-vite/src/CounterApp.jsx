import { useState } from 'react'
import PropTypes from 'prop-types'


export const CounterApp = ({value}) => {
    const [ counter, setCounter ] = useState(value);

    const sumar = () => {
        setCounter(counter + 1);
    }
    const restar = () => {
        setCounter(counter - 1);
    }
    const reiniciar = () => {
        setCounter(value);
    }

    return (
        <>
            <h1>CounterApp</h1>
            <h2>{counter}</h2>
            <button onClick={sumar}>+1</button>
            <button onClick={restar}>-1</button>
            <button aria-label="btn-reset" onClick={reiniciar}>Reset</button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
}