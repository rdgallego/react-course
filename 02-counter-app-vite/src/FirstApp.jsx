/* const newMessage = () => {
    return 'Roberto'
}; */

import PropTypes from 'prop-types';

export const FirstApp = ({title, subtitle}) => {
    if(!title){
        throw new Error('Titulo no existe')
    }
    return (
        <>
            <h1 data-testid="test-title"> {title} </h1>
            <p>{subtitle * 2}</p>
        </>
    )
}


FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.number.isRequired
}

FirstApp.defaultProps = {
    title: 'Hola, Soy Roberto',
}