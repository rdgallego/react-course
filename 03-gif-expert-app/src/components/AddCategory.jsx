import { useState } from "react"
import  PropTypes from "prop-types"

export const AddCategory = ({onAddCategory}) => {

    const [inputValue, setInputValue] = useState('One punch')
    const handleValue = ({target}) => {
        setInputValue(target.value)
    }
    const onSubmit = (event) => {
        event.preventDefault();
        if(inputValue.trim().length <= 1) return;
        onAddCategory(inputValue)
        setInputValue('');
    }
  return (
    <form aria-label="form" onSubmit={onSubmit}>
        <input type="text" placeholder="Buscar Gifs" value={inputValue} onChange={handleValue}></input>
    </form>
  )
}

AddCategory.propTypes = {
    onAddCategory: PropTypes.func.isRequired,
}