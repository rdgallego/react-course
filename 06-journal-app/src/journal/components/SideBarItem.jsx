import { TurnedInNot } from "@mui/icons-material"
import {  Grid, ListItem, ListItemButton, ListItemIcon, ListItemText } from "@mui/material"
import { useDispatch } from "react-redux"
import { setActiveNote } from "../../store/journal/journalSlice"

export const SideBarItem = ({note}) => {
    const dispatch = useDispatch()
    const activateNote = () => {
        dispatch(setActiveNote(note))
    }
  return (
    <ListItem disablePadding >
        <ListItemButton onClick={activateNote}>
            <ListItemIcon>
                <TurnedInNot/>
            </ListItemIcon>
            <Grid container>
                <ListItemText primary={note.title} />
                <ListItemText secondary={note.body} />
            </Grid>
        </ListItemButton>
    </ListItem>
  )
}
