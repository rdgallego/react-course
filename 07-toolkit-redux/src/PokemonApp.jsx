import { useEffect } from "react"
import { useDispatch, useSelector } from "react-redux"
import { getPokemons } from "./store/slices/pokemon/thunks"

export const PokemonApp = () => {
  const dispatch = useDispatch()
  const {isLoading, page, pokemons} = useSelector((state) => state.pokemons)
  useEffect(() => {
    return () => {
      dispatch(getPokemons())
    }
  }, [])
  

  return (
    <>
        <h1>Pokemon App</h1>
        <hr/>
        <span>Loading {isLoading ? 'True' : 'False'}</span>
        <ul>
            {pokemons.map((element) => {
              return <li key={element.name}>{element.name}</li>
            })}
        </ul>
        <button disabled={isLoading} onClick={() => {dispatch(getPokemons(page))}}>Next</button>
    </>
  )
}
