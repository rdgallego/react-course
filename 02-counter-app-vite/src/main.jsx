import React from 'react';
import ReactDOM from 'react-dom/client';
import { FirstApp } from './FirstApp';
/* import {CounterApp} from './CounterApp' */
import './styles.css'

ReactDOM.createRoot( document.getElementById('root')).render(
    <React.StrictMode>
        <FirstApp subtitle={123}/>
    </React.StrictMode>
)