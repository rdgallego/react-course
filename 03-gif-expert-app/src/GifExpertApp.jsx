import { useState } from "react"
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
    const [categories, setCategories] = useState([ 'One Punch', 'Dragon Ball' ]);
    const onAddCategory = (event) => {
        if(categories.includes(event)) return;
        setCategories([...categories, event])
    }
    return (
        <>
            <h1>GifExpert App</h1>
            <AddCategory onAddCategory={ onAddCategory }/>
            {
                categories.map(category =>
                    <GifGrid key={category} category={category}/>
                )
            }
        </>
    )
}