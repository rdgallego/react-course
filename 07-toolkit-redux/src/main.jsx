import React from 'react'
import ReactDOM from 'react-dom/client'
import App from './App.jsx'
import './index.css'
import { store } from './store/store.js'
import { Provider } from 'react-redux'
import { PokemonApp } from './PokemonApp.jsx'
import { TodoApp } from './TodoApp.jsx'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
     <Provider store={store}>
      <PokemonApp />
     </Provider>

  </React.StrictMode>,
)
