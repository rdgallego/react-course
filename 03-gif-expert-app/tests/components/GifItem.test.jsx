/* eslint-disable no-undef */
import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components"

describe('Pruebas archuvo GifItem', () => { 
    const title = 'saitama';
    const url = 'https://google.es/Saitama.jpg'

    test('should match snapshot', () => { 
        const {container} = render(<GifItem title={title} url={url}/>)
        expect(container).toMatchSnapshot();
     })

    test('should render image with url and alt', () => { 
        render(<GifItem title={title} url={url}/>)
        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
     })
     test('should show image title', () => { 
        render(<GifItem title={title} url={url}/>)
        expect(screen.getByText(title)).toBeTruthy();
      })
 })