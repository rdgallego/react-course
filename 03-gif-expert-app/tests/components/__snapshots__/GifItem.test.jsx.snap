// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas archuvo GifItem should match snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="saitama"
      src="https://google.es/Saitama.jpg"
    />
    <p>
      saitama
    </p>
  </div>
</div>
`;
